packages/storage/src/storage/useFiles.ts

```diff
@@ -52,16 +52,17 @@ const useFiles: UseFiles = ({ items, bucketName }) => {
   useEffect(() => {
     const fetchAndSetFiles = async ({ srcs }) => {
       const fetchedStorageUrls = await fetchStorageUrls({ srcs, bucketName })
-      const fetchedFiles = items.map((file, i) => ({
-        ...file,
-        url: fetchedStorageUrls[i],
-      }))
+      const fetchedFiles =
+        items?.map((file, i) => ({
+          ...file,
+          url: fetchedStorageUrls[i],
+        })) || []
       setFiles(fetchedFiles)
```

Comment:
if (!fetchedFiles?.length) return <<<<<<< just escape to prevent extra setState
setFiles(fetchedFiles)